src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1260,7 +1236,19 @@ void MarkEntireAssembly (AssemblyDefinition assembly)
 			MarkCustomAttributes (assembly.MainModule, new DependencyInfo (DependencyKind.AssemblyOrModuleAttribute, assembly.MainModule), null);
 
 			if (assembly.MainModule.HasExportedTypes) {
-				// TODO: This needs more work accross all steps
+				foreach (var exportedType in assembly.MainModule.ExportedTypes) {
+					_context.Annotations.Mark (exportedType, new DependencyInfo (DependencyKind.ExportedType, exportedType));
+
+					TypeDefinition type = exportedType.Resolve ();
+					if (type == null) {
+						if (!_context.IgnoreUnresolved)
+							_context.LogError ($"Exported type '{type.Name}' cannot be resolved", 1038);
```

Comment:
Some of the facades have platform-specific type forwards, so I'm worried this could introduce spurious errors. It may be better to ignore these resolution failures. We should at least check the warnings for a console app with "-p copy <forwarder>" for common forwarders like System.Runtime.

Reply:
I don't think ignoring them is correct behavior because it could generate invalid output. The error is ignored if you run with `--skip-unresolved`